src/messaging/ExchangeContext.cpp

```diff
@@ -411,11 +411,11 @@ CHIP_ERROR ExchangeContext::HandleMessage(const PacketHeader & packetHeader, con
 
         if (mDelegate != nullptr)
         {
-            mDelegate->OnMessageReceived(this, packetHeader, payloadHeader, std::move(msgBuf));
+            mDelegate->OnMessageReceived(this, packetHeader, payloadHeader, admin, std::move(msgBuf));
```

Comment:
should the admin id be part of the exchange context?

In my mind, the admin id depends on the peer that does the communication an exchange context tracks these exchanges. I am trying to reduce the number of arguments we pass to methods.

Reply:
The "admin" is not part of message layer. This is because:

- For PASE, there is no "admin" per-se: you have no authenticated identities, just peer-to-peer knowledge. 
- For Group messages, there is no "admin" either, just group ACLs, again because of no authenticated identities. 

Therefore, the "admin" concept is only relevant in CASE sessions, from metadata of CASE session establishment, and should be encoded in the peer data structures (which should be visible to OnMessageReceived)

Reply:
It sounds to me that we need some form of 'permission abstraction' instead of just an id.
I expect PASE to be equivaleent to some global admin (but no clear ID) and CASE to have an admin id.

I would also rather have this as a property of the exchange instead of a property of a single message (I imagine one cannot change identities within exchanges - you have to start a new exchange to do so)

Reply:
I agree that we should reduce the number of arguments, if possible. Let me find a way to do it. A valid admin ID is propagated only on message receive. So exchange context may not have this available if a node is just sending the messages.

Reply:
The updated PR removes the propagation of AdminId on message receive handler. It'll be stored in the SecureSessionHandle, which is owned by ExchangeContext. For now, I added an API to ExchangeContext to access the admin information (that contains the ACL). We can tweak it more once we settle on the final API for ACL and permissions.